### PR TITLE
Merge project setup with appropriate local dev environment settings

### PR DESCRIPTION
**Pull request by katporks**

### CODE REVIEW

**Comment by spwoodcock** [comment]
Looks good to me! Thanks for using the workflows I made 🙏 

I'm not sure how much you want me to meddle with your code/PR, so I will just leave comments 😄 

### Fix

I will update the `test_pytest` workflow to also read the Github env & secrets, similar to the test_compose workflow. Hopefully this will fix your issue.

The SECRET_KEY needs to be added to the github environment called `test` (the default, or you can name it whatever you want if you specify the arg in the workflow).

### Requirement

I could add an extra arg to pass the `example_dotenv_file`.

But in your case, I think it's simple to combine the two dotenv examples into one `.env.example`.
This file will be read by the workflow, environment variables substituted, and a .env file generated prior to running.

```dotenv
ENV=${ENV:-dev}
SECRET_KEY=${SECRET_KEY:-somesuperdupersecretkeyfortesting}

# SECURITY WARNING: define the correct hosts in production!
ALLOWED_HOSTS=${ALLOWED_HOSTS:-["*"]}

# Superuser creation (override in production)
DJANGO_SUPERUSER_USERNAME=${DJANGO_SUPERUSER_USERNAME:-admin}
DJANGO_SUPERUSER_EMAIL=${DJANGO_SUPERUSER_EMAIL:-admin@example.com}
DJANGO_SUPERUSER_PASSWORD=${DJANGO_SUPERUSER_PASSWORD:-adminpassword}

# Postgres DB settings
DB_NAME=${DB_NAME:-""}
DB_USER=${DB_USER:-""}
DB_PASSWORD=${DB_PASSWORD:-""}
DB_HOST=${DB_HOST:-""}
DB_PORT=${DB_PORT:-5432}
```

> SECRET_KEY has a default in the above config, so you wouldn't need to set it for testing.
> You could also set SECRET_KEY=${SECRET_KEY} with no default, to make sure it's always included (and doesn't accidentally use the default!).

### Alternatives

I am assuming your tests may require a database to run?

If this is the case, then the `test_compose` workflow may actually be more suited.
It runs a docker compose service, along with all other required services for it to operate (Django needs a DB).

### A Note on Dependencies

Also, it's personal choice, but I would really advise against using `pipenv`.
It's a pretty bad dependency manager compared to others.

Some good breakdowns:
https://chriswarrick.com/blog/2023/01/15/how-to-improve-python-packaging/
https://alpopkes.com/posts/python/packaging_tools/
https://frostming.com/2021/03-26/pm-review-2021/

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+.PHONY: build-dev build-prod test up-dev up-prod down-dev down-prod
+
+build-dev:
+	@docker-compose -f docker-compose.dev.yml build
```
Comment (dakotabenjamin) [inline Makefile:4]: recent versions of docker-compose use `docker compose` without the `-` so this is breaking in my local tests.

Comment (spwoodcock) [reply Makefile:4]: Docker compose V2 is much better (`docker compose`). It's Golang / much faster & has plugins inbuilt like buildkit.

Comment (katporks) [reply Makefile:4]: @dakotabenjamin @spwoodcock  gotcha! I'll switch to `docker compose`. thanks!

**Review by spwoodcock — APPROVED** [review]
Looks good 👍 

I merged the two dockerfiles into one, using multiple stages & added a few efficiency gains.
Give it a final check over to see if everything works as expected =)

You could rename `docker-compose.dev.yml` to `docker-compose.yml` if you want.
This way it will run as the default compose file with a simple `docker compose up -d`.

**Pull request merged**